tianshou/data/buffer.py

```diff
@@ -359,6 +360,78 @@ def __getitem__(
             policy=self.get(index, "policy"),
         )
 
+    @classmethod
+    def _copy_to_hdf5(
+            cls, k: str, v: Union[np.ndarray, Batch], grp: h5py.Group
+            ) -> None:
+        if isinstance(v, np.ndarray):
+            grp.create_dataset(k, data=v)
+        elif isinstance(v, torch.Tensor):
+            grp.create_dataset(k, data=v.numpy())
+        elif isinstance(v, Batch):
+            subgrp = grp.create_group(k)
+            for bk, bv in v.__dict__.items():
+                cls._copy_to_hdf5(bk, bv, subgrp)
+
+    @classmethod
+    def _copy_from_hdf5(
+            cls, grp: h5py.Group, dst: Batch, device: Optional[str] = "numpy"
+            ) -> None:
+        for k, v in grp.items():
+            if isinstance(v, h5py.Group):
+                if k not in dst.__dict__:
+                    dst.__dict__[k] = Batch()
+                cls._copy_from_hdf5(v, dst.__dict__[k])
+            elif isinstance(v, h5py.Dataset):
+                if k in dst.__dict__:
+                    if isinstance(dst.__dict__[k], np.ndarray):
+                        v.read_direct(dst.__dict__[k])
+                    elif isinstance(dst.__dict__[k], torch.Tensor):
+                        dst.__dict__[k] = torch.tensor(v, device=device)
+                    else:
+                        raise Exception("Cannot copy HDF5 dataset into object"
+                                        f"with type {type(dst.__dict__[k])}.")
+                else:
+                    if device == "numpy":
+                        dst.__dict__[k] = np.empty(v.shape, dtype=v.dtype)
+                        v.read_direct(dst.__dict__[k])
+                    else:
+                        dst.__dict__[k] = torch.tensor(v, device=device)
+
+    def save(self, path: str) -> None:
```

Comment:
I personally suggest changing this function name to `save_hdf5` because we may add other types of save/load functionality in the future. Same as load.

Reply:
Ok, good point. I have added _hdf5 to the function names.